components/scream/src/share/atm_process/atmosphere_process.hpp

```diff
@@ -60,9 +61,26 @@ namespace scream
 class AtmosphereProcess : public ekat::enable_shared_from_this<AtmosphereProcess>
 {
 public:
-  using TimeStamp       = util::TimeStamp;
-  using ci_string       = ekat::CaseInsensitiveString;
-  using ci_string_pair  = std::pair<ci_string,ci_string>; // Group and grid names
+  using TimeStamp      = util::TimeStamp;
+  using ci_string      = ekat::CaseInsensitiveString;
+
+  // A group request allow to request a group of fields.
+  struct GroupRequest {
+    GroupRequest (const std::string& name_, const std::string& grid_, const int ps = 1)
+     : name(name_), grid(grid_), pack_size(ps)
+    {
+      EKAT_REQUIRE_MSG(pack_size>=1, "Error! Invalid pack size request.\n");
+    }
+
+    // Group name
+    ci_string name;
+    // Grid name
+    ci_string grid;
+    // Request an allocation that can accomodate a value type like Pack<Real,pack_size>
+    int       pack_size;
+
+    friend bool operator<(const GroupRequest&, const GroupRequest&);
```

Comment:
Is this just for sorting requests?

Reply:
Yes, we need to overload op< so that we can stuff GroupRequest's inside an std::set. I'll add a comment so it's clear why it's there.

Reply:
Actually, since it's a struct, everything is public, so this friendship declaration is superfluous/misleading. I'll purge it, and put the comment above the op< declaration.